src/node/mod.rs

```diff
@@ -61,22 +64,41 @@ pub async fn node_with_autodiscovery(addresses_override: Option<String>) -> anyh
     Ok(())
 }
 
-fn parse_socket_list(socket_list: String) -> Vec<SocketAddr> {
-    socket_list
-        .split(',')
-        .map(|s| SocketAddr::from_str(s).unwrap())
-        .collect::<Vec<SocketAddr>>()
+fn parse_socket_list(socket_list: Vec<String>) -> Result<Vec<SocketAddr>, anyhow::Error> {
+    let (sockets, errors): (Vec<Result<_, _>>, Vec<Result<_, _>>) = socket_list
```

Comment:
Sorry to be so picky, but can we make this a `for {}` loop?

IMO it will make it more readable and you can abort the loop early using `.with_context()?` without having to iterate twice. I can change it if you want.

Reply:
No, that's perfect. I started with an iteration because I wanted to warn without panicking and it made sense. Then I decided that it wasn't worth it and we should panic without pinpointing the precise address that we could not parse. But at that point it became garbled so I'm happy to simplify it.